### ISSUE TITLE
Query: SELECT subquery order condition is ignored when filtering

### ISSUE THREAD

**Issue by gedsol** [aspnet/EntityFramework#4955]
The following code yields different results when executing against DBCollection and in-memory List of entities:

```C#
var query = ctx.Client.Select(c => new {
    client = c,
    ordersSorted = c.Orders.OrderByDescending(o => o.Date)
})
.Select(d => new {
    ClientId = d.client.Id,
    ClientName = d.client.Name,
    LastOrderDate = d.ordersSorted.FirstOrDefault().Date, // This works as intended
    LastCompletedOrderDate = d.ordersSorted.FirstOrDefault(o => o.Status == 5).Date, // Selected using OUTER APPLY without ORDER BY
    LastCancelledOrderDate = d.ordersSorted.FirstOrDefault(o => o.Status == 6).Date
});
var results = query.ToList();
```

**Comment by anpete** [comment]
Verified that we now generate the Order By.

**Comment by divega** [comment]
@anpete then it got fixed for 1.1.0-preview1?

**Comment by anpete** [comment]
@divega Not sure when exactly, the bug is pretty old.
